beets/library.py

```diff
@@ -790,6 +792,8 @@ class Album(LibModel):
         'original_year':      types.PaddedInt(4),
         'original_month':     types.PaddedInt(2),
         'original_day':       types.PaddedInt(2),
+
+        'data_source':        types.STRING,
```

Comment:
I would actually lean toward keeping the `data_source` field a flexible attribute rather than making a database column for it. It seems like a perfect scenario where we don't need this to be baked into the model and we could use the added flexibility.